src/vmm/src/device_manager/mmio.rs

```diff
@@ -156,38 +156,44 @@ impl MMIODeviceManager {
         }
     }
 
-    /// Allocates resources for a new device to be added.
-    fn allocate_new_slot(&mut self, irq_count: u32) -> Result<MMIODeviceInfo> {
-        let irqs = self.irqs.get(irq_count)?;
-        let slot = MMIODeviceInfo {
-            addr: self.next_avail_mmio,
-            len: MMIO_LEN,
-            irqs,
-        };
-        self.next_avail_mmio += MMIO_LEN;
-        Ok(slot)
+    // Allocates an irq and registers it with KVM for a given VM, event.
+    fn allocate_irq(&mut self, vm: &VmFd, evt: &EventFd) -> Result<Vec<u32>> {
+        let irq = self.irqs.get(1)?;
+        vm.register_irqfd(evt, irq[0])
+            .map_err(Error::RegisterIrqFd)?;
+        Ok(irq)
+    }
+
+    // Allocates slot and returns its base.
+    fn allocate_slot(&mut self, len: u64) -> Result<u64> {
+        if len < 0x100 {
+            return Err(Error::InvalidInput);
+        }
```

Comment:
Where does the `0x100` limit come from?

Reply:
It came from [this line](https://github.com/firecracker-microvm/firecracker/blob/8101086bbb942ca7accfd4f6be863fabf72f3d9b/src/vmm/src/device_manager/mmio.rs#L77)